helix-view/src/document.rs

```diff
@@ -150,13 +156,63 @@ pub struct Document {
     diff_handle: Option<DiffHandle>,
 }
 
+/// Inlay hints for a single `(Document, View)` combo.
+///
+/// There are `*_inlay_hints` field for each kind of hints an LSP can send since we offer the
+/// option to style theme differently in the theme according to the (currently supported) kinds
+/// (`type`, `parameter` and the rest).
+///
+/// Inlay hints are always `InlineAnnotation`s, not overlays or line-ones: LSP may choose to place
+/// them anywhere in the text and will sometime offer config options to move them where the user
+/// wants them but it shouldn't be Helix who decides that so we use the most precise positioning.
+///
+/// The padding for inlay hints is stored directly before/after the hint text itself. This is
+/// necessary to ensure it's placed correctly instead of after all the time (or ignored because it
+/// came in a later layer and at the same `char_idx`).
+#[derive(Debug, Clone)]
+pub struct DocumentInlayHints {
+    /// Identifier for the inlay hints stored in this structure. To be checked to know if they have
+    /// to be recomputed on idle or not.
+    pub id: DocumentInlayHintsId,
+
+    /// Inlay hints of `TYPE` kind, if any.
+    pub type_inlay_hints: Rc<[InlineAnnotation]>,
```

Comment:
We should probably be using `Arc` everywhere. I've avoided `Rc`

Reply:
In this case the only reasons this is a `Rc` instead of a `Vec` is that the virtual text API currently expects that instead of a slice. There are some other limitations imposed by using an Rc so I might just switch backto a slice, see https://github.com/helix-editor/helix/pull/5934#discussion_r1103890555.

I think there were some lifetime confusions initially in #5340 but I think those were resolved otherwise and I just kept the Rc to make the API easier to use.

Reply:
Resolving in favor of https://github.com/helix-editor/helix/pull/5934#discussion_r1103890555, easier for me to track one comment than two